content/en/blog/_posts/clientcmd-tutorial.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: blog
+title: 'Uniform API server access using clientcmd'
+date: 2024-07-24
+slug: clientcmd-tutorial
+author: Stephen Kitt (Red Hat)
+---
+
+If you've ever wanted to develop a command-line client for a Kubernetes API,
+especially if you’ve considered making your client usable as a `kubectl` plugin,
+you might have wondered how to make your client feel familiar to users of `kubectl`.
+A quick glance at the output of `kubectl options` might put a damper on that:
+“am I really supposed to implement all those options?”
+
+Fear not, others have done a lot of the work involved for you.
+In fact, the Kubernetes provides two libraries to help you handle
+`kubectl`-style command-line options,
+[`clientcmd`](https://pkg.go.dev/k8s.io/client-go/tools/clientcmd) and
+[`cli-runtime`](https://pkg.go.dev/k8s.io/cli-runtime)
+(which uses `clientcmd`).
+This article will show how to use the former.
+
+## General philosophy
+
+As might be expected since it’s part of `client-go`,
+`clientcmd`’s ultimate purpose is to provide an instance of
+[`restclient.Config`](https://pkg.go.dev/k8s.io/client-go/rest#Config)
+that can issue requests to an API server.
+
+It follows `kubectl` semantics:
+* defaults are taken from `~/.kube` or equivalent;
+* files can be specified using the `KUBECONFIG` environment variable;
+* all the above can be further overridden using command-line arguments.
+
+One major missing piece for full `kubectl`-style argument handling is support
```

Comment:
```suggestion
One major missing piece for complete `kubectl` style argument handling is the support
```

Reply:
I’ve rephrased it:

```markdown
It doesn’t set up a `--kubeconfig` flag,
which you might want to do to align with `kubectl`;
you’ll see how to do this [later on](#bind-the-flags).
```

Reply:
Thats good. Maybe you can rewrite the last one as `you'll see how to do this in [bind-the-flags] section later in this blog` Current one is ok as well. It just add some more direct reference here.